aptos-move/move-examples/staking/sources/oracle.move

```diff
@@ -0,0 +1,50 @@
+module staking::oracle {
+    use aptos_std::math128;
+    use pyth::i64;
+    use pyth::price;
+    use pyth::price_identifier;
+    use pyth::pyth;
+
+    friend staking::commission;
+
+    const PRECISION: u128 = 100000000; // 1e8
+    const INITIAL_MAX_AGE_SECS: u64 = 120; // 2 minutes
+    const PYTH_APT_ID: vector<u8> = x"03ae4db29ed4ae33d323568895aa00337e658e348b37509f5372ae51f0af00d5";
+    const PYTH :vector<u8> = b"pyth";
+
+    /// Price read from oracle is stale
+    const ESTALE_PRICE: u64 = 1;
+
+    struct OracleConfig has key {
+        max_age_secs: u64,
+    }
+
+    fun init_module(staking_signer: &signer) {
+        move_to(staking_signer, OracleConfig {
+            max_age_secs: INITIAL_MAX_AGE_SECS,
+        });
+    }
+
+    #[view]
+    public fun get_apt_price(): u128 acquires OracleConfig {
+        let config = &OracleConfig[@staking];
+        let price = pyth::get_price_no_older_than(price_identifier::from_byte_vec(PYTH_APT_ID), config.max_age_secs);
+        let raw_price = i64::get_magnitude_if_positive(&price::get_price(&price));
+        let expo = price::get_expo(&price);
+        // Standardize precision or otherwise we'll get different magnitudes for different decimals
+        math128::mul_div(
+            (raw_price as u128),
+            PRECISION,
+            math128::pow(10, (i64::get_magnitude_if_negative(&expo) as u128)),
+        )
+    }
+
+    public fun precision(): u128 {
+        PRECISION
+    }
```

Comment:
Any reason not to inline?

Is that because it's a constant?  probably would be better to just inline and put the value here rather than have a public function

Reply:
Inlined